tests/helpers.test.ts

```diff
@@ -0,0 +1,108 @@
+     writeFileSync(settingsFilePath, JSON.stringify(expectedSettings));
+
+     const actualSettings = helpers.readSettings(serviceName, settingsFilePath);
+
+     expect(actualSettings).toEqual(expectedSettings);
+   });
+
+   test("Verify cannot load settings.json because it does not exist", () => {
+     //eslint-disable-next-line no-console
+     console.log = jest.fn();
+     const serviceName = "Settings";
+     unlinkSync(settingsFilePath);
+
+     const actualSettings = helpers.readSettings(serviceName, settingsFilePath);
+
+     //eslint-disable-next-line no-console
+     expect(console.log).toHaveBeenCalledWith(expect.stringContaining("[Settings] Unable to read the configuration file: ENOENT: no such file or directory"));
+     expect(actualSettings).toBeNull();
+   });
+
+   test("Verify throws if rawConfig empty", () => {
+     const serviceName = "Settings";
+     const expectedSettings = "";
+     writeFileSync(settingsFilePath, expectedSettings);
+
+     expect(() => {helpers.readSettings(serviceName, settingsFilePath)}).toThrow(Error);
+   });
+
+   test("Verify throws with message if rawConfig empty", () => {
+     const serviceName = "Settings";
+     const expectedSettings = "";
+     writeFileSync(settingsFilePath, expectedSettings);
+
+     try {
+       helpers.readSettings(serviceName, settingsFilePath);
+     } catch (error) {
+       expect(error.message).toBe(`[${serviceName}] Unable to load configuration file ${settingsFilePath}.`);
+     }
+   });
+
+  test("Verify throws if json invalid", () => {
+    const serviceName = "Settings";
+    const expectedSettings = {};
+    writeFileSync(settingsFilePath, JSON.stringify(expectedSettings));
+    const mockAddListener = jest.spyOn(JSONC, 'parse');
+    mockAddListener.mockImplementation((rawConfig, parseErrors) => {
+        parseErrors.push({
+            error: 1,
+            offset: 2,
+            length: 3,
+        });
+        return {};
+    });
+
+    expect(() => {helpers.readSettings(serviceName, settingsFilePath)}).toThrow(Error);
+  });
+
+  test("Verify throws with message if json invalid", () => {
+    const serviceName = "Settings";
+    const expectedSettings = {};
+    writeFileSync(settingsFilePath, JSON.stringify(expectedSettings));
+    const mockAddListener = jest.spyOn(JSONC, 'parse');
+    mockAddListener.mockImplementation((rawConfig, parseErrors) => {
+        parseErrors = [{
+            error: 1,
+            offset: 2,
+            length: 3,
+        }];
+        return {};
+    });
+    //eslint-disable-next-line no-console
+    console.log = jest.fn();
+
+     try {
+       helpers.readSettings(serviceName, settingsFilePath);
+     } catch (error) {
+       //eslint-disable-next-line no-console
+       expect(console.log).toHaveBeenCalledWith(expect.stringContaining("[Settings] 1"));
+       expect(error.message).toBe(`[${serviceName}] Unable to load configuration file: `);
+     }
+  });
+});
```

Comment:
I based a lot of this testing on some previous work I did here: https://github.com/TonyBrobston/logs-to-mqtt-publisher/blob/master/tests/index.test.ts

Reply:
Do you have the Prettier extension installed in VSCode? That'll make sure the style matches everything else automatically.

Reply:
I'm developing in vim.

Also I really just meant like, `test` vs. `it`, test name convention, test file name convention, etc.

Reply:
Ah good old vim. I used to have a vi reference coffee mug.

Can you please run `npm run format` then and push any test files that get changed? Thanks!

Reply:
The next time I'm in front of my computer I'll do this 🙂.

Reply:
@danecreekphotography  Ok, I ran that.